### DISCUSSION TITLE
Ideas / Examples of "quick start" and other useful code snippets for coffea 2023.

### DISCUSSION THREAD

**Discussion by lgray** [scikit-hep/coffea#939] [Ideas]
Using this thread to collect code snippets from folks that can be included into coffea 2023 documentation.

**Comment by btovar** [comment]
Here is a draft for taskvine. I'm using `pip install --pre coffea` so that the examples run. (Making a note to remember to update on release.)
[tvqs.md](https://github.com/CoffeaTeam/coffea/files/13663327/tvqs.md)

**Comment by btovar** [comment]
From old coffea:
```python
runner = processor.Runner(exec_instance, schema=NanoAODSchema, chunksize=chunksize, maxchunks=nchunks)
```
schema should be set in `NanoEventsFactory`.
`chunksize` and `maxchunks` do not exists anymore. 
`chunksize` closest equivalent in `NanoEventsFactory` is `steps_per_file`, but coffea warns that it should only be used in small tests.

**Reply by lgray** [reply]
Not quite the case!

`max_chunks` is now handled with a function, and it operates on dataset definitions rather than being a parameter of the analysis itself.

https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/dataset_tools/manipulations.py#L12

You can see it being used in the following test:
https://github.com/CoffeaTeam/coffea/blob/master/tests/test_dataset_tools.py#L189

There's also the more general `slice_chunks` function now.

**Reply by lgray** [reply]
So you specify the chunks in the `steps` section of the dataset definition, and then operate upon that.
You could also create by hand a test dataset that that only specifies some single, small step to process.

**Reply by lgray** [reply]
And you specify `chunksize` now through the separate `preprocess` function that spits out a completely specified preprocessed dataset.

You can see its use here:
https://github.com/CoffeaTeam/coffea/blob/master/tests/test_dataset_tools.py#L204
